test/arcade/core/util/ParameterTest.java

```diff
@@ -0,0 +1,150 @@
+
+        Parameter p = new Parameter(mu, sigma, false, random);
+
+        for (int i = 0; i < iterations; i++) {
+            double value = p.nextDouble();
+            assertTrue(value >= minValue);
+            assertTrue(value <= maxValue);
+        }
+    }
+
+    @Test
+    public void nextInt_positiveMu_returnsWithinRange() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        double mu = 10.0;
+        double sigma = 0.5;
+        double maxValue = 20;
+        double minValue = 0;
+        int iterations = 10000;
+
+        Parameter p = new Parameter(mu, sigma, false, random);
+
+        for (int i = 0; i < iterations; i++) {
+            int value = p.nextInt();
+            assertTrue(value >= minValue);
+            assertTrue(value <= maxValue);
+        }
+    }
+
+    @Test
+    public void nextInt_negativeMu_returnsWithinRange() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        double mu = -4.0;
+        double sigma = 0.25;
+        double maxValue = -2;
+        double minValue = -6;
+        int iterations = 10000;
+
+        Parameter p = new Parameter(mu, sigma, false, random);
+
+        for (int i = 0; i < iterations; i++) {
+            int value = p.nextInt();
+            assertTrue(value >= minValue);
+            assertTrue(value <= maxValue);
+        }
+    }
+
+    @Test
+    public void nextDouble_isFrac_staysWithin0and1() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        double mu = 0.5;
+        double sigma = 1;
+        int iterations = 10000;
+
+        Parameter p = new Parameter(mu, sigma, true, random);
+
+        for (int i = 0; i < iterations; i++) {
+            double value = p.nextDouble();
+            assertTrue(value >= 0.0);
+            assertTrue(value <= 1.0);
+        }
+    }
+
+    @Test
+    public void constructor_isFracAndMuGreaterThan1_throwsIllegalArgumentException() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        double mu = 2.0;
+        double sigma = 0.25;
+
+        IllegalArgumentException e =
+                assertThrows(
+                        IllegalArgumentException.class,
+                        () -> {
+                            new Parameter(mu, sigma, true, random);
+                        });
+        assertEquals("Fractional parameter must be between 0 and 1", e.getMessage());
+    }
+
+    @Test
+    public void constructor_isFracAndMuLessThan0_throwsIllegalArgumentException() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        double mu = -2.0;
+        double sigma = 0.25;
+
+        IllegalArgumentException e =
+                assertThrows(
+                        IllegalArgumentException.class,
+                        () -> {
+                            new Parameter(mu, sigma, true, random);
+                        });
+        assertEquals("Fractional parameter must be between 0 and 1", e.getMessage());
```

Comment:
I don't love testing for the exact exception message. Thoughts on something like a custom `OutOfBoundsException` that extends `IllegalArgumentException`?

Reply:
I feel like I'm adding an unnecessary class to maintain for a single use exception, but I'll add if a project goal is more robust error handling during runtime.